state/connection/ClientState.kt

```diff
@@ -48,7 +49,10 @@ import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
 import mu.KotlinLogging
 
-class ClientState constructor(private val notificationMgr: NotificationManager) {
+class ClientState constructor(
+    private val notificationMgr: NotificationManager,
+    private val preferenceMgr: PreferenceManager
```

Comment:
We need to supply the preference manager down this route along with the notification manager. I think there is a case for just supplying `StudioState` as a single argument here.

Reply:
We shouldn't supply `StudioState` here. Because `StudioState` is too broad. If we want to refactor the arguments, introduce a `ClientContext`. But if that seems like a thin abstraction (which I think it does) then we just keep the arguments.

There is precedent for just keeping the arguments already: see `state.project.ProjectManager`.

Reply:
Alright, kept as is.